src/main/java/com/javaproject/restservice/repository/UserRepository.java

```diff
@@ -0,0 +1,11 @@
+package  com.javaproject.restservice.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+import com.javaproject.restservice.entity.User;
+
+
+@RepositoryRestResource(path = "/users")
```

Comment:
What does this do? @RepositoryRestResource?

Reply:
Also why does this need to be an interface?
What is JpaRepository?
and why is is User, String?

Reply:
The @RepositoryRestResource annotation marks this interface as a REST resource for /users

The interface extends JpaRepository and  The interface also provides all the CRUD implementations and create a Spring bean.

There are  two generic types defined when interface is extended  entity class and its primary key. My entity class is User and the type is String.
 